### ISSUE TITLE
Platforms and stops gets removed from existing relation

### ISSUE THREAD

**Issue by Hidooo** [Zaczero/osm-relatify#36]
I have tried out the tool on a couple of routes, but either platforms or stops seem to get removed from the relation after uploading the changeset to OSM. I also can't click on both `platform` and `stop_position` for the same bus stop, it seems to be either or. This unfortunately makes the tool unusable on routes that are somewhat mapped already, but just need a bit of correction of the ways. The relations in question are [2919652](https://www.openstreetmap.org/relation/2919652) (fixed by me afterwards, but you can see all the platforms/stop_positions that relatify removed in the history) and [2726495](https://www.openstreetmap.org/relation/2726495) (not fixed by me afterwards).
Am I doing something wrong or is there a bug in the tool? I compared two bus stops that were tagged identically, and one of them ([11151650](https://www.openstreetmap.org/relation/11151650)) somehow got both `platform` and `stop_position` included in the relation, while the other ([11165954](https://www.openstreetmap.org/relation/11165954)) only got the `stop_position` included.

**Comment by Zaczero** [comment]
1) Cases like those are only supported, when <kbd>highway=bus_stop</kbd> is mapped off the road:

![image](https://github.com/Zaczero/osm-relatify/assets/10835147/9c1373a2-b14f-4d41-b83a-51328b79fdf1)

2) When discussing cases such as the one at https://www.openstreetmap.org/way/164510362, the primary issue causing Relatify to struggle lies in its treatment of <kbd>highway=bus_stop</kbd> as the most significant feature of all. This level of importance attributed is much more explicit compared to the platform mentioned. It's worth noting that once a <kbd>highway=bus_stop</kbd> tag is identified, it is implicitly regarded as <kbd>public_transport=platform</kbd>, even when it represents a stop_position. This decision is made to accommodate some commonly used but poorly tagged instances, which account for a substantial portion (16%).

![image](https://github.com/Zaczero/osm-relatify/assets/10835147/f0db63d9-9c07-4b5b-8db4-978d2665b558)

I can only suggest an improvement in bus tagging in this area. Here is some useful documentation:
- https://wiki.openstreetmap.org/wiki/Tag:highway%3Dbus_stop
- https://wiki.openstreetmap.org/wiki/Tag:public%20transport=platform
- https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dstop_position
- https://wiki.openstreetmap.org/wiki/Tag:route%3Dbus

In summary, I would recommend recommend to follow the <kbd>highway=bus_stop</kbd> documentation which states:

> The [highway](https://wiki.openstreetmap.org/wiki/Key:highway)=bus_stop tag is widely used on a [node](https://wiki.openstreetmap.org/wiki/Node) off to one side of the highway way to identify the position where passengers wait for a bus beside the carriageway.

Having <kbd>highway=bus_stop</kbd> at the same node as <kbd>public_transport=stop_position</kbd> is slightly confusing and should be improved on (passengers do not wait on the road).

**Comment by Zaczero** [comment]
Let me provide a good example:
- https://www.openstreetmap.org/node/7356621639

Ideally, https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dstop_area could be added here to remove any duplicated information.

**Comment by Hidooo** [comment]
Hmm, after having reread all public transport documentation on the wiki, it seems to be a systemic problem in the whole of Sweden, stemming from this wiki article: [https://wiki.openstreetmap.org/wiki/Sweden/Public_transport#Hur_skall_vi_tagga_nu?](https://wiki.openstreetmap.org/wiki/Sweden/Public_transport#Hur_skall_vi_tagga_nu?). From what I can tell, it was discussed how to upgrade old tagging to the then (2013) new public transport tagging schema, and people tried out different methods, and subsequently settled on the method described there. It contradicts international wiki pages on public transport tagging, and especially considers `highway=bus_stop` to be a legacy tag, secondary to public transport tagging and `stop_area` relations.

However, it seems to be a problem not only in Sweden but also mainly in Germany and a region in France. Maybe it needs to be handled after all? Or should we push for a mass retagging of PT infrastructure in these countries?
![image](https://github.com/Zaczero/osm-relatify/assets/38591400/1e244e20-6579-44ad-9da8-a6169a80ad85)
([Overpass Ultra query](https://overpass-ultra.trailsta.sh/#query=/*%0AThis%20has%20been%20generated%20by%20the%20overpass-turbo%20wizard.%0AThe%20original%20search%20was:%0A%E2%80%9Chighway=bus_stop%20and%20public_transport=stop_position%E2%80%9D%0A*/%0A%5Bout:json%5D%5Btimeout:2500%5D;%0A//%20gather%20results%0A(%0A%20%20//%20query%20part%20for:%20%E2%80%9Chighway=bus_stop%20and%20public_transport=stop_position%E2%80%9D%0A%20%20node%5B%22highway%22=%22bus_stop%22%5D%5B%22public_transport%22=%22stop_position%22%5D(%7B%7Bbbox%7D%7D);%0A);%0A//%20print%20results%0Aout%20body;%0A%3E;%0Aout%20skel%20qt;&map=4.672047581078059/53.231269/8.615113))

TLDR: Looks like public transport tagging in Sweden is currently incompatible with relatify and that will not change in the near future.

**Comment by Zaczero** [comment]
If you examine the regions near Stockholm (https://www.openstreetmap.org/relation/3879443), you will observe instances of (unofficial) PTv3 tagging, a rather unconventional selection by the Swedish community. It seems reasonable to adopt tagging that adheres to global compatibility. However, based on my observations, this principle doesn't seem to consistently apply there. Nevertheless, they have the right to choose their approach, and that should be respected. I simply wish for more collaboration with other countries on their part. Their PTv3 implementation seems intentionally designed to NOT maintain backward compatibility with the existing PTv2 standard. Adapting Relatify to function with it would demand significant efforts.

**Comment by Zaczero** [comment]
Having said that, let me mark this as wontfix.

**Comment by Zaczero** [comment]
> However, it seems to be a problem not only in Sweden but also mainly in Germany and a region in France. Maybe it needs to be handled after all? Or should we push for a mass retagging of PT infrastructure in these countries?

The tagging that your Overpass query examines **is indeed supported** (https://github.com/Zaczero/osm-relatify/issues/36#issuecomment-1684915736). The initial issue you raised is slightly different, as I explained during. Kindly review my response and feel free to ask if any aspects remain unclear.
